tests/cli/bmccreds/test_main.py

```diff
@@ -160,7 +213,71 @@ def test_random_password_with_domain(self):
             xnames=self.mock_hsm_client.get_and_filter_bmcs.return_value,
             domain='bmc',
             force=False,
-            report_format='pretty'
+            report_format='pretty',
+            length=RANDOM_PASSWORD_LENGTH,
+            allowed_chars=VALID_CHAR_SETS_STRING
+        )
+
+    def test_random_password_with_password_length(self):
+        """Supplying --random-password with a password length"""
+        self.cli_args = ['bmccreds', '--random-password', '--password-length', '8']
+        do_bmccreds(self.parse_args())
+        self.mock_get_username_and_password.assert_not_called()
+        self.bmccreds_cls.assert_called_once_with(
+            password=None,
+            xnames=self.mock_hsm_client.get_and_filter_bmcs.return_value,
+            domain=None,
+            force=False,
+            report_format='pretty',
+            length=RANDOM_PASSWORD_LENGTH,
+            allowed_chars=VALID_CHAR_SETS_STRING
+        )
+
+    def test_random_password_with_password_char_sets(self):
+        """Supplying --random-password with a password character set"""
+        self.cli_args = ['bmccreds', '--random-password', '--password-char-sets', 'alpha,numeric,symbols']
+        do_bmccreds(self.parse_args())
+        self.mock_get_username_and_password.assert_not_called()
+        self.bmccreds_cls.assert_called_once_with(
+            password=None,
+            xnames=self.mock_hsm_client.get_and_filter_bmcs.return_value,
+            domain=None,
+            force=False,
+            report_format='pretty',
+            length=RANDOM_PASSWORD_LENGTH,
+            allowed_chars=VALID_CHAR_SETS_STRING
+        )
+
+    def test_random_password_with_password_chars(self):
+        """Supplying --random-password with explicit password characters"""
+        self.cli_args = ['bmccreds', '--random-password', '--password-chars', 'abcdefgh']
+        do_bmccreds(self.parse_args())
+        self.mock_get_username_and_password.assert_not_called()
+        self.bmccreds_cls.assert_called_once_with(
+            password=None,
+            xnames=self.mock_hsm_client.get_and_filter_bmcs.return_value,
+            domain=None,
+            force=False,
+            report_format='pretty',
+            length=RANDOM_PASSWORD_LENGTH,
+            allowed_chars='abcdefgh'
+        )
+
+    def test_random_password_with_password_length_and_password_char_sets(self):
+        """Supplying --random-password with password length and password character set"""
+        self.cli_args = ['bmccreds', '--random-password', '--password-length', '8', '--password-char-sets',
+                         'alpha,numeric,symbols'
```

Comment:
Probably okay to leave this one as you have it, since it's not bad to test that all three charsets work as expected.

Reply:
Here, I left as it is to test all three charsets